### ISSUE TITLE
input type=range onChange should fire when changing the value using the keyboard arrow keys

### ISSUE THREAD

**Issue by subtleGradient** [facebook/react#554]

**Comment by dmmalam** [comment]
+1

Any workaround for IE10+ to get react to fire a synth onChange, on a browser onchange for range inputs

**Comment by chooven** [comment]
+1 for input on IE10+ firing onChange

**Comment by kwhitaker** [comment]
+1 to all the +1

**Comment by luisrudge** [comment]
any news? range input is still broken in IE

**Comment by rippo** [comment]
+1 for this to be applied. Doesn't work for me even on Edge

**Comment by TheDeveloperXYZ** [comment]
Does not work in IE 11.

**Comment by JeanCarriere** [comment]
+1

**Comment by bsudekum** [comment]
For what it's worth, I created a work around component while this is worked out in core: https://github.com/mapbox/react-range

**Comment by dancoates** [comment]
+1, Love the work that everyone is doing on react, but I do feel like this bug should be a higher priority to fix seeing as we are approaching the 2 year mark from its discovery.

**Comment by spicyj** [comment]
@dancoates I'm sorry that this is a lower priority for us than for you. Feel free to send a pull request though! `ChangeEventPlugin` is a little hairy but my blog post at http://benalpert.com/2013/06/18/a-near-perfect-oninput-shim-for-ie-8-and-9.html should give you an idea of the general strategy. For range elements, we should listen to both input and change events and fire events when either happens (but only once even if both happen).

**Comment by dancoates** [comment]
No need to be sorry, I realise it is a tricky one. I'll take a look at ChangeEventPlugin and see if I can help.
Perhaps a caveats section in the docs would help folks wondering why changes aren't firing? Though I guess this thread is easy enough to find.

**Comment by raphaelparent** [comment]
:+1: would love to see this fixed!

**How I dealt with the problem for now**
For my use, I simply added the same event on a `onMouseUp` event and it works for all IE 10 and above and Edge. 

It doesn't give the exact same result because the value is only changed on the `mouseup` event but I think (for my case anyway) that it degrades beautifully.

**Comment by zanarmstrong** [comment]
I also ran into this & would love to see it fixed.

Thanks for the suggestions for workarounds. Adding an "onClick" or "onMouseup" event was very helpful, but it's still suboptimal that nothing happens until the user let's go of the slider.

**Comment by kevinfrei** [comment]
Check out [this](http://stackoverflow.com/questions/18544890/onchange-event-on-input-type-range-is-not-triggering-in-firefox-while-dragging) on StackOverflow which explains a way to address getting updates while the user is dragging...

**Comment by AlexReff** [comment]
Why is this closed? This is still a problem - onChange does not fire at all in IE11 in our testing. Forced to use https://github.com/mapbox/react-range

**Comment by jvanderberg** [comment]
We are using 15.3.2, and are experiencing this issue.  Looking at the release notes for the three minor releases since then, I don't see that this has been fixed.  Why is it closed?  Issue https://github.com/facebook/react/issues/8168 seems to be the tracking case for it, but it seems like this issue has a much more thorough description of the problem.

**Comment by gaearon** [comment]
If you look at the history above, you will see that the issue was closed by #5746.
If you open this PR, you will see that its milestone is set to 16.

So the fix will be in 16, and we can’t put it in 15.x because it introduces a breaking change in behavior.

I don’t know if it’s possible to backport it to 15.x. @jquense and @nhunzaker would likely be able to answer this.

**Comment by jquense** [comment]
It is possible! There is even a PR :) already  #8575

**Comment by gaearon** [comment]
Oh nice. Can you make a list of DOM PRs that are hanging in there awaiting review?
Then add them to umbrella in #8583 so that the relevant cases have manual tests.

**Comment by aweary** [comment]
https://github.com/facebook/react/pull/8575#issuecomment-269679723 is on my todo list. Just need to test some stuff locally and then I think we'll be 👍 to merge. I'll make sure to add some relevant cases to #8583 when I get a chance to review it

**Comment by nhunzaker** [comment]
Anyone working on a browser test fixture for this? If not, I've got some time to write one up.

**Comment by jquense** [comment]
I was planning on it but haven had a moment yet. feel free to jump in

**Comment by nhunzaker** [comment]
No worries. There's plenty of other test cases to write up. I've got a PR out related to https://github.com/facebook/react/issues/8308. I'll shift gears to that.

**Comment by c0debreaker** [comment]
I encountered this issue today on my Windows phone. It does not call the method I specified in onChange. When is the fix going to be applied?

**Comment by gaearon** [comment]
This has been released in 15.6.x.
